docs/src/how-to/install/ingress.md

```diff
@@ -0,0 +1,102 @@
+
+*at the time of writing (2023-03), this section assumes you use a kubernetes
+version 1.23 or above (tested with 1.26)*
+
+## Installing in a cloud-like environment
+
+Install the ingress controller chart in your helmfile with the defaults, simply
+like this:
+
+```yaml
+# helmfile.yaml
+repositories:
+  - name: wire
+    url: 'https://s3-eu-west-1.amazonaws.com/public.wire.com/charts'
+
+releases:
+  - name: 'ingress-nginx-controller'
+    namespace: 'wire'
+    chart: 'wire/ingress-nginx-controller'
+    version: 'CHANGE_ME'
+
+# charts wire-server and nginx-ingress-services also need to be installed, see other
+# documentation
+#  - name: ...
+#    chart: ...
+```
+
+By default, the `wire/ingress-nginx-controller` chart will create a `Deployment`
+with services of type `LoadBalancer`, where your kubernetes installation needs
+to support dynamic LoadBalancers. If this is not possible, read the next section.
+
+By default three pods will come up and external traffic will be load balanced into these
+three pods, which will also do TLS termination and forward traffic to upstream
+services (`nginz` and others).
+
+To inspect default TLS settings, see [defaults in the latest code](https://github.com/wireapp/wire-server/blob/develop/charts/ingress-nginx-controller/values.yaml) and also see {ref}`tls`.
+
+## Installing on bare-metal without dynamic load balancer support
+
+In case you cannot create a `kind: service` of `type: LoadBalancer`, then you
+can fall back to manually ensure traffic reaches your installation:
+
+```yaml
+# helmfile.yaml
+releases:
+  - name: 'ingress-nginx-controller'
+    namespace: 'wire'
+    chart: 'wire/ingress-nginx-controller'
+    version: 'CHANGE_ME'
+    values:
+      - './helm_vars/ingress-nginx-controller/values.yaml'
+```
+
+Create this file with the following override values:
+
+```yaml
+# helm_vars/ingress-nginx-controller/values.yaml
+ingress-nginx:
+  controller:
+    kind: DaemonSet
+    service:
+      type: NodePort
+```
+
+Then, on each of your kubernetes worker nodes, two ports are exposed: ports
+31773 (https) and 31772 (http)
+
+You should add a port-forwarding rule on the node or on the loadbalancer that
+forwards ports 443 and 80 to these respective ports. Any traffic hitting the http port is simply getting a http 30x redirect to https.
+
+Downsides of this approach: The NodePort approach always requires manual configuration of some external load balancer/firewall to round-robin between node IPs and is error-prone. It's also a bit annoying to have to decide on some global ports that may not be used otherwise.
+
+Most managed K8s clusters have support for LoadBalancers, you can also get this for your own clusters in hcloud etc. It's even possible to do it for pure bare metal, without any "load balancer hardware", by using BGP or some leadership election over who's announcing the "load balancer ip" via ARP (https://metallb.universe.tf/configuration/_advanced_l2_configuration/).
```

Comment:
This is okay for a github discussion, but a bit too sloppy for public docs.

What about

> Most managed K8s clusters have support for LoadBalancers.
> Manually set up K8S Clusters can also support this, by using a provider/environment-specific CCM
>  (see [hcloud](https://github.com/hetznercloud/hcloud-cloud-controller-manager) and [digitalocean](https://github.com/digitalocean/digitalocean-cloud-controller-manager) for examples).
> In case you're provisioning on bare metal, without any hardware load balancer support in front,
> you might be using [MetalLB](https://metallb.universe.tf/), which supports BGP or Failover L2 ARP announcements.
> The choice of CCM highly depends on the environment you choose to deploy wire-server in.